### PR TITLE
Apply `max_returned_metrics` from prometheus annotations to openmetrics check

### PR DESCRIPTION
**Pull request by khewonc**
### What does this PR do?

Allows the `max_returned_metrics` parameter to be configured from prometheus annotations.

### Motivation

Support ticket

### Describe how to test your changes

* Spin up a pod with prometheus k8s annotations
* Set up [metric collection with prometheus k8s annotations on the agent](https://docs.datadoghq.com/agent/kubernetes/prometheus/#metric-collection-with-prometheus-annotations)
* Set `DD_PROMETHEUS_SCRAPE_CHECKS` with `max_returned_metrics`:
```yaml
  prometheusScrape:
    enabled: true
    additionalConfigs:
      - configurations:
          - timeout: 5
            send_distribution_buckets: true
            max_returned_metrics: 2500
```
* Check the openmetrics configuration to verify `max_returned_metrics` is set with `kubectl exec -it <agent-pod> -c agent -- agent configcheck`:
```
=== openmetrics check ===
Configuration provider: prometheus-pods
Configuration source: prometheus_pods:docker://<cid>
Instance ID: openmetrics:<id>
max_returned_metrics: 2500
metrics:
- '*'
...
```

### Checklist

- [x] A [release note](https://github.com/DataDog/datadog-agent/blob/main/docs/dev/contributing.md#reno) has been added or the `changelog/no-changelog` label has been applied.
- [ ] The `need-change/operator` and `need-change/helm` labels has been applied if applicable.
- [ ] The [config template](https://github.com/DataDog/datadog-agent/blob/main/pkg/config/config_template.yaml) has been updated if applicable.

Note: Adding GitHub labels is only possible for contributors with write access.

### CODE REVIEW

**Review by ahmed-mez — APPROVED** [review]
thank you!

File: releasenotes/notes/add-max-returned-metrics-option-for-prometheus-annotations-a0aa2141f014de72.yaml [hunk 1]
```diff
@@ -0,0 +1,12 @@
+# Each section from every release note are combined when the
+# CHANGELOG.rst is rendered. So the text needs to be worded so that
+# it does not depend on any information only available in another
+# section. This may mean repeating some details, but each section
+# must be readable independently of the other.
+#
+# Each section note must be formatted as reStructuredText.
```
Comment (sgnn7) [inline releasenotes/notes/add-max-returned-metrics-option-for-prometheus-annotations-a0aa2141f014de72.yaml:7]: Nit: You don't need the preamble

**Review by sgnn7 — APPROVED** [review]
Tiny non-blocking nit but LGTM

**Comment by sgnn7** [comment]
@khewonc  Looks like the release note filename length is too long too

**Pull request merged**